app/src/test/java/ar/com/wolox/android/training/ui/training/login/LoginPresenterTest.java

```diff
@@ -0,0 +1,110 @@
+package ar.com.wolox.android.training.ui.training.login;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import ar.com.wolox.android.training.utils.CredentialsSession;
+
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class LoginPresenterTest {
+
+    private static final String EMAIL_INVALID = "testInvalidEmail.com";
+    private static final String NON_EXISTENT_EMAIL = "test@test.com";
+    private static final String EMAIL = "melvin.lambert15@example.com";
+    private static final String PASSWORD = "qwerty";
+
+    private ILoginView iLoginView;
+    private CredentialsSession credentials;
+
+    private LoginPresenter presenter;
+    private LoginAdapter adapter;
+
+    @Before
+    public void createInstances() {
+        iLoginView = mock(ILoginView.class);
+        credentials = mock(CredentialsSession.class);
+        adapter = mock(LoginAdapter.class);
+
+        presenter = new LoginPresenter(credentials, adapter);
+    }
+
+    @Test
+    public void testInstancesCreated() {
+        Assert.assertNotNull(presenter);
+    }
+
+    @Test
+    public void testAlreadyUserStoredInSharedPreferences() {
+        doReturn(EMAIL).when(credentials).getUsername();
```

Comment:
You're testing nothing here! You're saying that when calling `credentials.getUsername()` it returns EMAIL, and then you test that. You aren't testing the logic you developed.